params/protocol_params.go

```diff
@@ -261,17 +261,35 @@ func SstoreClearsScheduleRefund(stateSize, contractSize *big.Int) uint64 {
 }
 
 func CalculateGasWithStateScaling(stateSize, contractSize *big.Int, baseRate uint64) uint64 {
-	var scalingFactor *big.Int
+
+	if stateSize == nil {
+		stateSize = big.NewInt(0)
+	}
+	if contractSize == nil {
+		contractSize = big.NewInt(0)
+	}
+
+	stateTrieLevels := big.NewInt(0)
+	contractTrieLevels := big.NewInt(0)
+	// First calculate the log2 * 2^64 of the state size and contract size and
+	// add them together
 	if stateSize.Sign() != 0 {
```

Comment:
We should check to make sure this is > 1

Reply:
changed